models/lido/liquidity/ethereum/lido_liquidity_ethereum_curve_steth_ng_pool.sql

```diff
@@ -0,0 +1,233 @@
+where call_success = TRUE 
+)
+
+
+, wsteth_rate as (
+SELECT
+  day, rate as rate0, value_partition, first_value(rate) over (partition by value_partition order by day) as rate,
+  lead(day,1,date_trunc('day', now() + interval 1 day)) over(order by day) as next_day
+  
+FROM (
+select day, rate,
+sum(case when rate is null then 0 else 1 end) over (order by day) as value_partition
+from (
+select  date_trunc('day', d.day) as day, 
+        sum(cast(steth as double))/sum(cast(wsteth as double)) AS rate
+from dates  d
+left join volumes v on date_trunc('day', v.time)  = date_trunc('day', d.day) 
+group by 1
+))
+
+)
+
+, steth_in as (
+select
+    DATE_TRUNC('day', evt_block_time) as time,
+    sum(cast(value as double))/1e18 as steth_in,
+    sum(cast(value as double)/coalesce(r.rate, 1))/1e18 as wsteth_in,
+    r.rate
+from {{source('erc20_ethereum','evt_Transfer')}} t
+ left join wsteth_rate r on DATE_TRUNC('day', evt_block_time) >= r.day and DATE_TRUNC('day', evt_block_time) < r.next_day
+where 
+    contract_address = lower('0xae7ab96520de3a18e5e111b5eaab095312d7fe84') and 
+    to = lower('0x21e27a5e5513d6e65c4f830167390997aa84843a') and
+    DATE_TRUNC('day', evt_block_time) >= to_date('{{ project_start_date }}')
+    
+group by 1,4
+)
+
+, steth_out as (
+select
+    DATE_TRUNC('day', evt_block_time) as time,
+    -sum(cast(value as double))/1e18 as steth_in,
+    -sum(cast(value as double)/coalesce(r.rate, 1))/1e18 as wsteth_in,
+    rate
+from {{source('erc20_ethereum','evt_Transfer')}} t
+ left join wsteth_rate r on DATE_TRUNC('day', evt_block_time) >= r.day and DATE_TRUNC('day', evt_block_time) < r.next_day
+where 
+    contract_address = lower('0xae7ab96520de3a18e5e111b5eaab095312d7fe84') and 
+    from = lower('0x21e27a5e5513d6e65c4f830167390997aa84843a') and
+    DATE_TRUNC('day', evt_block_time) >= to_date('{{ project_start_date }}')
+    
+group by 1, 4
+)
+
+, daily_balances as (
+select time, sum(steth_in) steth_balance, sum(wsteth_in) as wsteth_balance from (
+select * from steth_in
+union all
+select * from steth_out
+) group by 1
+)
+
+, steth_balances as (
+select time, 
+lead(time, 1, now()+ interval 1 day ) over (order by time) as next_time,
+sum(steth_balance) over (order by time) as steth_cumu,
+sum(coalesce(wsteth_balance,steth_balance)) over (order by time) as wsteth_cumu, r.rate,
+(sum(coalesce(wsteth_balance,steth_balance)) over (order by time))*coalesce(r.rate, 1) as steth_from_wsteth 
+from daily_balances b
+left join wsteth_rate r on b.time >= r.day and b.time < r.next_day 
+order by 1
+)
+
+, eth_balances as (
+select time, lead(time, 1, date_trunc('day', now() + interval 1 day)) over (order by time) as next_time, sum(eth_amount) over (order by time) as eth_pool
+from (
+SELECT date_trunc('day', block_time) as time, SUM(amount) as eth_amount
+FROM (
+    -- outbound transfers
+    SELECT block_time, from AS address, -cast(tr.value as double)/1e18 AS amount
+    FROM {{source('ethereum','traces')}} tr
```

Comment:
can all three base tables here in the union be filtered by the `{{ project_start_date }}`?